Plugins/Public/alley/PlayerRestrictions.cpp

```diff
@@ -997,62 +1023,48 @@ bool ExecuteCommandString_Callback(CCmds* cmds, const wstring &wscCmd)
 	return false;
 }
 
-void __stdcall HkCb_AddDmgEntry_AFTER(DamageList *dmg, unsigned short p1, float& damage, enum DamageEntry::SubObjFate fate)
+void __stdcall HkCb_AddDmgEntry(DamageList *dmg, unsigned short p1, float& damage, enum DamageEntry::SubObjFate& fate)
 {
 	returncode = DEFAULT_RETURNCODE;
-	if (iDmgToSpaceID && dmg->get_inflictor_id() && dmg->is_inflictor_a_player())
-	{
-		uint client = HkGetClientIDByShip(iDmgToSpaceID);
-		if (client)
-		{
-			uint ShootingClient = dmg->get_inflictor_owner_player();
-			Archetype::Ship* TheShipArch = Archetype::GetShip(Players[ShootingClient].iShipArchetype);
 
-			if (TheShipArch->iShipClass == 19)
-			{
-				float curr, max;
-				pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
-				float expecteddmg = (float)1;
-				float projecteddamage = curr - damage;
+	if (iDmgMunitionID != repairMunitionId
+		|| !iDmgTo
+		|| !dmg->is_inflictor_a_player()
+		|| p1 == 65521)
+		return;
 
-				//If a repair gun heals the ship but this doesn't show up, it's because it's hitting the shield.
-				//HkMsgU(L"DEBUG: damage by repair ship, is it healing?");					
-				//PrintUserCmdText(client, L"Projected damage: %f", projecteddamage);
+	if (!iDmgToSpaceID)
```

Comment:
why is this no longer triggering an immediate return?

Reply:
iDmgToSpaceID is null on external equipment and collision group hits. Im 'fixing' the plugin by making it possible to fix the ship regardless of where on he hull the hit landed as long as it was not a shield hit.
Very impactful when dealing with new Battlecruisers on which a large chunk of their surface area are collision groups.